src/Mono.Android/Java.Interop/TypeManager.cs

```diff
@@ -318,7 +318,7 @@ public static void RegisterType (string java_class, Type t)
 					if (jniFromType != java_class) {
 						managedToJni.Add (t, java_class);
 					}
-				} else {
+				} else if (!ignoreExistingRegistration) {
```

Comment:
Additionally, the only reason for the new parameter is to *not* emit a warning message (?).

Why not skip these `TypeManager`-related changes entirely and live with an additional warning? IIRC the log messages aren't even going anywhere "important" anyway...